server/controllers/companyController.js

```diff
@@ -92,516 +93,404 @@ const getCompanies = async (req, res) => {
+
+      if (found !== -1) {
+        //2. calculate tenure and add it to updateData correspondingly
+        let newDifference = 0;
+        let origDifference = 0;
+        //2a. calculate new tenure
+        if (!company.indefinite) {
+          const differenceMs =
+            new Date(company.endDate) - new Date(company.startDate); // calc diff in ms
+          // Convert the difference to days
+          newDifference = math.round(differenceMs / (1000 * 60 * 60 * 24));
+        } else if (
+          company.isIndefinite &&
+          new Date(company.startDate) < new Date()
+        ) {
+          // Calculate the difference in milliseconds
+          const differenceMs = math.abs(date2 - date1);
+
+          // Convert the difference to days
+          newDifference = math.round(differenceMs / (1000 * 60 * 60 * 24));
+        }
 
-        if (!response) {
-          res.status(404).json({ error: "company not found" });
+        //2b. calculate old tenure
+        if (!origCompanies[found].indefinite) {
+          const origDate1 = new Date(origCompanies[found].startDate);
+          const origDate2 = new Date(origCompanies[found].endDate);
+
+          const origDifferenceMs = origDate2 - origDate1;
+
+          origDifference = origDifferenceMs / (1000 * 60 * 60 * 24);
+        } else if (
+          company.isIndefinite &&
+          new Date(company.startDate) < new Date()
+        ) {
+          const origDate1 = new Date(origCompanies[found].startDate);
+          const origDate2 = new Date(); //Get new date
+          const origMs = origDate2 - origDate1;
+          origDifference = math.round(origMs / (1000 * 60 * 60 * 24));
         }
-      } else {
-        const response = await Company.updateOne(
-          { name: lowercaseCompanyName },
-          {
-            $inc: {
-              rating: (rating ? rating : 0) - (removeRating ? removeRating : 0),
-              tenure: tenure ? tenure : 0,
-            },
-            $pull: {
-              Employees: { $in: removeEmployees },
-              ratedEmployees: { $in: removeEmployees },
-              interns: { $in: removeEmployees },
-            },
-          }
-        );
 
-        if (!response) {
-          res.status(404).json({ error: "company not found" });
+        //2c. calculate the new total tenure and add to update data
+        const totalTenure = newDifference - origDifference;
+        updateData.$inc["tenure"] = totalTenure;
+
+        //3. calculate correct rating and add it to the update data
+        const totalRating = company.rating - origCompanies[found].rating;
+        updateData.$inc["rating"] = totalRating;
+
+        //4. Add employees
+        //4a. Decipher between interns and full time workers
+        const containsIntern = company.title.includes("intern");
+        if (containsIntern) {
+          updateData.$addToSet["interns"] = company.userId;
+        } else {
+          updateData.$addToSet["Employees"] = company.userId;
         }
```

Comment:
Great documentation / comments, WAY easier to read and understand what's going on now.